servicecontrol/upgrades/3to4/index.md

```diff
@@ -0,0 +1,103 @@
+```
+
+If the ServiceControl instance being upgraded manages an audit queue, then additional parameters must be specified for the creation of a new ServiceControl Audit instance.
+
+```ps
+Invoke-ServiceControlInstanceUpgrade `
+  -Name <Name of main instance> `
+  -InstallPath <Path for Audit instance binaries> `
+  -DBPath <Path for the Audit instance database> `
+  -LogPath <Path for the Audit instance logs> `
+  -Port <Port for the Audit instance api> `
+  -DatabaseMaintenancePort <Port for the Audit instance embedded database> `
+  [-ServiceAccountPassword <password for service account>] `
+  [-Force]
+```
+The following information is copied from the existing ServiceControl instance:
+
+- Audit queue
+- Audit log queue
+- Forward audit messages
+- Audit retention period
+- Transport
+- Connection string
+- Host name
+- Service account
+
+NOTE: If this instance uses a domain account, the the account password must be supplied.
+
+The name of the new audit instance will be derived from the name of the original instance.
+
+
+## Upgrading multi-instance deployments
+
+Upgrading a multi-instance ServiceControl deployment must be done in stages. Some stages may require the use of the powershell scripts. 
+
+### Upgrade the primary instance
+
+The first step is to upgrade the primary ServiceControl instance. If the primary instance has audit ingestion enabled, then a new ServiceControl Audit instance will be created for it.
+
+NOTE: Once the primary instance has been upgraded, it will not subscribe to events being published by new secondary instances. All subscriptions to existing secondary instances will be retained. As the primary instance no longer requires the transport address of the secondary instances to send subscription requests, this property has been dropped from the `ServiceControl/RemoteInstances` configuration setting.
+
+### Upgrade the secondary instances
+
+Once the primary instance has been upgraded to version 4, secondary instances can be upgraded one at a time. If a secondary instance has audit ingestion enabled, then a new ServiceControl Audit instance will be created for it.
+
+If a secondary instance has error ingestion turned off, then it cannot be upgraded to version 4. The recommended course of action is to create a side-by-side deployment with a new ServiceControl Audit instance until the audit retention period cleans up the messages in the original secondary instance. Follow this sequence of steps:
+
+- Note the name of the primary instance
+- Check the configuration file of the secondary instance and note the following values:
+  - Transport
+  - Connection string
+  - Audit queue
+  - Audit log queue
+  - Audit forwarding
+  - Audit retention period
+- Turn off audit ingestion on the secondary instance by setting `ServiceBus/AuditQueue` to `!disable`. 
+- Create a new ServiceControl Audit instance with configuration that matches the settings in the original instance
```

Comment:
I wonder if we could make this whole procedure a bit smarter now that we merged the endpoint detail type forwarding binder. For example when I do

```
Get-ServiceControlInstances | Where-Object { $_.Name -eq "InstanceName" }
```

I get this object

```
Name                    : Particular.ServiceControl.9
Url                     : http://localhost:33357/api/
HostName                : localhost
Port                    : 33357
DatabaseMaintenancePort : 33358
InstallPath             : C:\sc\Particular.ServiceControl.9
DBPath                  : C:\sc\db\Particular.ServiceControl.9
LogPath                 : C:\sc\logs\Particular.ServiceControl.9
TransportPackageName    : RabbitMQ - Conventional routing topology
ConnectionString        : host=localhost
ErrorQueue              : error9
ErrorLogQueue           : error9.log
ForwardErrorMessages    : False
ServiceAccount          : LocalSystem
Version                 : 3.8.2
```

so technically we could use this and feed it into `New-AuditInstance` then the user doesn't have to note down anything. Weird is though that because of the refactoring we did the AuditQueue information is no longer there so we would need to reintroduce it but at least we can get ports, paths etc.